code_folder/22_ml_inference_overlay.py

```
@@ -0,0 +1,269 @@
+    return model
+
+
+# ---------------------------------------------------------------------------
+# 2. Run Inference
+# ---------------------------------------------------------------------------
+def run_inference(model, X_test: pd.DataFrame) -> tuple[np.ndarray, np.ndarray]:
+    """Generate predictions and probabilities."""
+    print(f"[Inference] Running predictions on {X_test.shape[0]} samples")
+    y_pred = model.predict(X_test)
+    y_proba = model.predict_proba(X_test)[:, 1]
+    print(f"  -> Predictions generated")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)